### DISCUSSION TITLE
Update software list format and inclusion criteria

### DISCUSSION THREAD

**Discussion by reece** [HGVSnomenclature/hgvs-nomenclature#97] [Other discussion topics]
@ifokkema and @Peter-J-Freeman proposed updates to the software list in https://github.com/HGVSnomenclature/hgvs-nomenclature/discussions/56#discussioncomment-6743111. This discussion thread is to continue that discussion toward the next version of the software list. Possible topics include:

- Principles for inclusion (e.g., OSS only?, commercial?, license options, kinds of functionality)
- Functionality categories and subcategories, and perhaps whether we assess them
- Re-enable descriptions, and whether to impose criteria on style
- Whether to include github stats, links to documentation, etc.

**Comment by jfjlaros** [comment]
In response to [this](https://github.com/HGVSnomenclature/hgvs-nomenclature/commit/f220d5a08801853f972c02aaa4be370341b7a7d2?notification_referrer_id=NT_kwDOAHCFOrI4NTEzODQ2NTQzOjczNzQxMzg#commitcomment-133447572) comment,

> I would like to pare the list down to categories that represent the major factors that influence decisions

Support for allele descriptions is a major factor in my opinion, as is support for RNA and protein descriptions.

> does r. as output constitute handling RNA variants?

I would think not. To handle, again in my opinion, should include accepting these variant descriptions as input and being able to validate them. Ideally, a tool or library should be able to validate its own output.

**Reply by Peter-J-Freeman** [reply]
- I would think not. To handle, again in my opinion, should include accepting these variant descriptions as input and being able to validate them. Ideally, a tool or library should be able to validate its own output.

I agree. Handling r. means being able to accept and r., validate it and output an r. 

- Support for allele descriptions is a major factor in my opinion, as is support for RNA and protein descriptions.

Agree here too. Sure, I know there are ongoing discussion around when to merge variants or not, but handling the input of an allele and giving valid outputs is important.

**Reply by Peter-J-Freeman** [reply]
I like the idea of github stats e.g. build status, codecov docs etc. I know I have work to do to get mine up to scritch, but these are useful.

One thing we need to think about for inclusion is "how do we determine whether a solution is sufficiently robust". For example, we could include VEP, but IMHO, it is not robust and makes huge errors. Alamut is very good, but does not handle genomic reference sequences well, but I would generally trust it.


For functionality, perhaps @reece @jfjlaros and I should compise a list of what we consider to be functionality, and discuss what to include from the lists. @ifokkema could moderate :)

> Re-enable descriptions, and whether to impose criteria on style
Can you expand on this @reece

**Comment by reece** [comment]
> For functionality, perhaps @reece @jfjlaros and I should compise a list of what we consider to be functionality, and discuss what to include from the lists. @ifokkema could moderate :)

I think we already have the list, we just don't agree on the level of detail or categories. However, I also really don't care as much as all of this. Would anyone care to either 1) start a discussion or 2) craft a PR with a proposed change? If a PR, the narrower the better.
 
> > Re-enable descriptions, and whether to impose criteria on style
> Can you expand on this @reece

Yep. We used to have descriptions in the software table. Johan was concerned that they sounded like an advertisement, and Alex wanted criteria for what might go in to them. I just wanted to get a minimalist table out, so I dropped the description column for expediency.

**Reply by Peter-J-Freeman** [reply]
Ah, makes sense. For me, I had no issues with dropping it for a minimal table. I think Johan is right r.e. sounding like adverts. Well, they sort of are. So, a better way to go in my opinion is linking to documentation, even if it is just the Git README or other. No need for us to provide this in the table

**Comment by ifokkema** [comment]
I would like to include the LOVD HGVS DNA variant description syntax checker, but it doesn't have its own publication yet. However, the NCBI Variation Services mentioned now don't link to a publication mentioning those services now either, and all links to GitHub point to a stale unrelated repo so I guess I can include my service?

On categories and descriptions; I understand the rationale for dropping the descriptions, but we might want to have a place to add details about the categories. E.g, the API may have less functionality than the web UI or the service validates variant descriptions but not on the sequence level.

**Reply by Peter-J-Freeman** [reply]
> the API may have less functionality than the web UI or the service validates variant descriptions but not on the sequence level.
Or indeed more functionality.

This is why it may potentially make more sense to split the tools so that the APIs get an additional entry.  Means we can keep the format neat without bloat. Also means we can recycle existing functionality. Again, I think the majority of the detail should come from the documentation for the resource
